packages/blade/src/components/Table/_decisions/decisions.md

```diff
@@ -0,0 +1,543 @@
+
+```jsx
+// Mock data & functions
+const data = [{
+    firstName: 'John',
+    lastName: 'Doe',
+    balance: 1000
+},
+{
+    firstName: 'Jane',
+    lastName: 'Doe',
+    balance: 2000
+}];
+
+
+const sortFunctions = {
+  firstName: (array) => array.sort((a, b) => a.name.localeCompare(b.name)),
+};
+
+const onSelectionChange = (selectedItems) => {
+  console.log(selectedItems);
+};
+
+const columnDefinition = [
+    {
+        header: 'First Name',
+        headerKey: 'firstName',
+        cell: (row) => row.firstName,
+        isSticky: true,
+        columnStart: undefined,
+        columnEnd: undefined,
+    },
+    {
+        header: 'Last Name',
+        headerKey: 'lastName',
+        cell: (row) => row.lastName,
+        isSticky: true,
+        columnStart: undefined,
+        columnEnd: undefined,
+    },
+    {
+        header: () => (
+            <>
+                Balance
+                <Tooltip content='Current bank balance'>
+                    <InfoIcon size='medium' />
+                </Tooltip>
+            </>),
+        headerKey: 'balance',
+        cell: (row) => (
+            <>
+                {row.balance}
+                <Badge>Low</Badge>
+            </>),
+        isSticky: true,
+        columnStart: undefined,
+        columnEnd: undefined,
+    }
+];
+
+const footerDefinition = [
+    {
+        footer: 'Total',
+        columnStart: 1,
+        columnEnd: 4,
+    },
+    {
+        footer: computedTotalBalance,
+    }
+];
+
+const pagination = {
+  showRowCountPicker?: boolean,
+  onRowCountPickerChange?: ({ count }) => void,
+  navigationType?: 'compact' | 'expanded',
+  label?: string,
+  labelPosition?: 'left' | 'right',
+  limit: number,
+  offset: number,
+  onPaginationChange?: ({ offset }) => void,
+};
+```
+
+```jsx
+// Compact Table
+
+<Table
+    data={data} // This contains the actual data to be rendered in the table which would be retrieved from some API
+    columns={columns} // This contains the column definition for the table
+    footerRow={footerRow} // This contains the footer definition for the table
+    selectionType='single'
+    onSelectionChange={onSelectionChange}
+    sortFunctions={sortFunctions}
+    onSortChange={onSortChange}
+    isLoading={false}
+    cellDensity='normal'
+    showZebraStripes={true}
+    toolbarTitle="Users"
+    toolbarItemsSelectedTitle={`${selectedUsersCount} Users selected`}
+    toolbarActions={<Button>Export</Button>}
+    isStickyHeader={true}
+    isStickyFooter={true}
+/>
+```
+
+##### Pros
+- Compact API is more concise and requires less effort to create a table
+- Most popular table libraries like AG Grid, React Table, etc. use a compact API which is battle tested and developers are more familiar with the same
```

Comment:
Has React table etc. used compact API to reduce verbosity only or there are other reasons ?

Reply:
Using APIs that define columns and layout instead of components that compose columns and layouts easier when you need to support multiple frameworks which is what react table does